flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -1029,6 +1029,8 @@ private TableResultInternal executeInternal(
                         defaultJobName,
                         jobStatusHookList);
         try {
+            ClassLoader userClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
Please add coments and refer to the v2 implementation in the comments and that the v2 refactor is not going to be backported to 1.20.

Reply:
Added comments on the need for this change. The latest FLIP implementation to introduce a stream graph based job submission moved the StreamGraph module to flink runtime and changed the job submission logic by directly submitting a StreamGraph to the job manager. 
Ref FLIP: https://cwiki.apache.org/confluence/display/FLINK/FLIP-468%3A+Introducing+StreamGraph-Based+Job+Submission
Related JIRA: https://issues.apache.org/jira/browse/FLINK-36065
As a result of these changes this issue seems to be fixed for flink v2 as I was not able to reproduce it with the latest main. Hence limiting this particular change to 1.20 versions.